### DISCUSSION TITLE
Bug: Images are not shown anymore

### DISCUSSION THREAD

**Discussion by MuslemRahimi** [pocketbase/pocketbase#2338] [Q&A] [answered]
Hi,

I have this weird bug in "production mode" when i create a new "post" for my post collection in sveltekit that image of the post are not shown anymore.

I looked at pocketbase to see if the image was stored which it is. At my "dev mode" everything works fine where i did not change anything from my code. 

This bug appeared a couple of days ago. I believe when i switch to version 0.15 (but not sure if this is the cause).

Is anyone else have this problem?

Any help is much appreciated

**Comment by ganigeorgiev** [comment]
The description is too vague and there aren't enough details to help you debug it.

Are you using S3 or the local filesystem?
How are those images uploaded, from the TinyMCE `editor` field or via a dedicated `file` field?
If it is from within a TinyMCE `editor` fields, new images are not "uploaded" anywhere and are embedded as base64 text as part of your content.

Please provide an example or at least the actual request error (you can find it in the browser's Inspect tool > Network and/or Console tabs).

**Reply by MuslemRahimi** [reply]
Sorry, it is kinda hard for me to explain.

I use a local filesystem and also no TinyMCE is involved.

I uploaded an example image here: https://stocknear.com/community

On my local machine it works fine and the image is shown. The image also exist in the database (see attachment)

![Screenshot from 2023-04-22 09-52-58](https://user-images.githubusercontent.com/22890183/233770858-a1385d79-6113-4306-8085-923af2898b94.png)

**Reply by MuslemRahimi** [reply]
What is also weird is when i make the same post on my local machine and transfer the pb_data to production i can see the image on my website.

**Reply by ganigeorgiev** [reply]
Could you open the browser's Inspect tool and check what is the image request error?

Do you have a reverse proxy (nginx, apache, caddy, etc.) that could interfere with the regular file serving endpoint? Are there any error logs? Could you share your configuration?

**Reply by MuslemRahimi** [reply]
I use a nginx reverse proxy with the config

```
##############Sveltekit Stuff##############

server {



	root /var/www/html;

	# Add index.php to the list if you are using PHP
	index index.html index.htm index.nginx-debian.html;

	server_name stocknear.com www.stocknear.com;

	location / {
		# First attempt to serve request as file, then
		# as directory, then fall back to displaying a 404.
		proxy_pass http://localhost:3000;
    		proxy_http_version 1.1;
    		proxy_set_header Upgrade $http_upgrade;
    		proxy_set_header Connection 'upgrade';
    		proxy_set_header Host $host;
		proxy_cache_bypass $http_upgrade;
	}

}



server {
    if ($host = www.stocknear.com) {
        return 301 https://$host$request_uri;
    } # managed by Certbot


    if ($host = stocknear.com) {
        return 301 https://$host$request_uri;
    } # managed by Certbot


	listen 80 default_server;
	listen [::]:80 default_server;

	server_name stocknear.com www.stocknear.com;
    return 404; # managed by Certbot


}


##############Pocketbase Stuff##############
server {
    listen 80;
    server_name stocknear.com;
    client_max_body_size 10M;

    location / {
        # check http://nginx.org/en/docs/http/ngx_http_upstream_module.html#keepalive
        proxy_set_header Connection '';
        proxy_http_version 1.1;
        proxy_read_timeout 360s;

        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;

        # enable if you are serving under a subpath location
        # rewrite /yourSubpath/(.*) /$1  break;

        proxy_pass http://127.0.0.1:8090;
    }
}

```

**Reply by MuslemRahimi** [reply]
The inspector tool gives me
`<img src="http://127.0.0.1:8090/api/files/zwcqzkgkgiesllb/tj59keqhaw25the/screenshot_from_2023_04_22_09_52_58_AZ6B0eotDl.png?thumb=0x0" alt="image" class="shadow-xl relative m-auto object-center w-auto h-full md:h-96 rounded-xl">`

![Screenshot from 2023-04-22 10-30-34](https://user-images.githubusercontent.com/22890183/233772901-e9131689-53b6-4f8b-b24c-7cc9f05902cb.png)


Sorry i cant give you better infos than this

**Reply by ganigeorgiev** [reply, marked as answer]
Why are you sending requests to the localhost address? Is this on production? If yes, make sure that you use the correct url address when initializing the JS SDK.

I don't understand the above nginx config (how are you serving both applications on the  same port for the same domain?), but if stocknear.com is where PocketBase is accessible, then you have to initialize the SDK with that url, aka:

```js
new PocketBase("https://stocknear.com")
```

**Reply by MuslemRahimi** [reply]
thanks to your input i fixed the issue :)
